lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -111,38 +84,98 @@ describe('NxTabs', function () {
       </NxTabs>
     );
 
-    expect(component.find('[role="tab"]').first()).not.toHaveClassName('active');
-    expect(component.find('[role="tab"]').last()).toHaveClassName('active');
-    expect(component.find('[role="tabpanel"]')).toHaveText('Content 1');
+    const selectedTabs = getAllByRole('tab', { selected: true });
+    expect(selectedTabs.length).toBe(1);
+    expect(selectedTabs[0]).toHaveTextContent(/^Tab 1/);
+    expect(getByRole('tabpanel')).toHaveTextContent('Content 1');
+  });
+
+  it('calls onTabSelect when a tab is clicked', async function() {
+    const user = userEvent.setup();
+    const onTabSelect = jest.fn();
+
+    const { getByRole } = quickRender({
+      activeTab: -1,
+      onTabSelect
+    });
+
+    await user.click(getByRole('tab'));
+
+    expect(onTabSelect).toHaveBeenCalledWith(0);
   });
 
-  it('calls onTabSelect when a tab is clicked', function () {
+  it('calls onTabSelect when a tab is selected via keyboard', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+
+    const { getAllByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect,
+      children: (
         <NxTabList>
           <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
         </NxTabList>
-      </NxTabs>
-    );
+      )
+    });
 
-    component.find('[role="tab"]').simulate('click');
+    const firstTab = getAllByRole('tab')[0];
 
-    expect(onTabSelect).toHaveBeenCalledWith(0);
+    firstTab.focus();
+    await user.keyboard('{ArrowRight}{Enter}');
+
+    expect(onTabSelect).toHaveBeenCalledWith(1);
   });
 
-  it('calls onTabSelect when a tab is selected via keyboard', function () {
+  it('calls onTabSelect when a tab is selected via keyboard', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+
+    const { getAllByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect,
+      children: (
         <NxTabList>
           <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
         </NxTabList>
-      </NxTabs>
-    );
+      )
+    });
 
-    component.find('[role="tab"]').simulate('keypress', { key: ' ' });
+    const firstTab = getAllByRole('tab')[0];
 
-    expect(onTabSelect).toHaveBeenCalledWith(0);
+    firstTab.focus();
+    await user.keyboard('{ArrowRight}{Enter}');
+
+    expect(onTabSelect).toHaveBeenCalledWith(1);
+  });
+
+  it('sets id and aria labelledby to panel to reference active tab aria-controls', async function() {
```

Comment:
This test seems like it could be simplified.  Rather than using a stateful fixture and user interaction, why not just do two renders (or a render and rerender) with different props?

Reply:
> This test seems like it could be simplified. Rather than using a stateful fixture and user interaction, why not just do two renders (or a render and rerender) with different props?

Yup, updated here: b51e5f8197b8378624b69466c5584a03f5fc9052